lib/structures/CommandInteraction.js

```diff
@@ -0,0 +1,382 @@
+
+const {InteractionResponseTypes} = require("../Constants");
+
+/**
+* Represents an application command interaction. See Interaction for more properties.
+* @extends Interaction
+* @prop {PrivateChannel | TextChannel | NewsChannel} channel The channel the interaction was created in. Can be partial with only the id if the channel is not cached.
+* @prop {Object} data The data attached to the interaction
+* @prop {String} data.id The ID of the Application Command
+* @prop {String} data.name The command name
+* @prop {Number} data.type The [command type](https://discord.com/developers/docs/interactions/application-commands#application-command-object-application-command-types)
+* @prop {String?} data.target_id The id the of user or message targetted by a context menu command
+* @prop {Array<Object>?} data.options The run Application Command options
+* @prop {String} data.options.name The name of the Application Command option
+* @prop {Number} data.options.type Command option type, 1-10
+* @prop {Number?} data.options.value The value of the run Application Command (Mutually exclusive with options)
+* @prop {Array<Object>?} data.options.options The run Application Command options (Mutually exclusive with value)
+* @prop {Object?} data.resolved converted users + roles + channels
+* @prop {Collection<User>?} data.resolved.users converted users
+* @prop {Collection<Member>?} data.resolved.members converted members
+* @prop {Collection<Role>?} data.resolved.roles converted roles
+* @prop {Collection<Channel>?} data.resolved.channels converted channels
+* @prop {String?} guildID The ID of the guild in which the interaction was created
+* @prop {Member?} member The member who triggered the interaction (This is only sent when the interaction is invoked within a guild)
+* @prop {User?} user The user who triggered the interaction (This is only sent when the interaction is invoked within a dm)
+*/
+class CommandInteraction extends Interaction {
+    constructor(info, client) {
+        super(info, client);
+
+        this.channel = this._client.getChannel(info.channel_id) || {
+            id: info.channel_id
+        };
+
+        this.data = info.data;
+
+        if(info.data.resolved !== undefined) {
+            //Users
+            if(info.data.resolved.users !== undefined) {
+                const usermap = new Collection(User);
+                Object.entries(info.data.resolved.users).forEach(([id, user]) => {
+                    usermap.set(id, new User(user, this._client));
+                });
+                this.data.resolved.users = usermap;
+            }
+            //Members
+            if(info.data.resolved.members !== undefined) {
+                const membermap = new Collection(Member);
+                Object.entries(info.data.resolved.members).forEach(([id, member]) => {
+                    member.user = {id};
+                    membermap.set(id, new Member(member, this._client.guilds.get(this.guildID), this._client));
```

Comment:
`guildID` is not defined at this stage. Causing the member's guild property to be undefined.

Reply:
oop I am stupid